### PR TITLE
git-range-diff: add page

### PR DESCRIPTION
**Pull request by xeruf**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

Helpful SO link for reference:
https://stackoverflow.com/questions/52323008/can-somebody-explain-the-usage-of-git-range-diff

### CODE REVIEW

File: pages/common/git-range-diff.md
Comment (marchersimon) [inline pages/common/git-range-diff.md:10]: I'm not sure what this command does, but I think there should be better words for `ours` and `theirs`. @bl-ue what do you think?

Comment (xerus2000) [reply pages/common/git-range-diff.md:10]: Please look at the SO question I linked for a longer explanation

Comment (bl-ue) [reply pages/common/git-range-diff.md:10]: Personally, I don't mind the usage of `ours` and `theirs` because that's official Git wording, and [we allow such exceptions](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md#inclusive-language).

Comment (bl-ue) [reply pages/common/git-range-diff.md:10]: However, we don't seem to use this terminology in any of the other Git pages, so maybe we should try to move away from it.  @xerus2000 also see the [`git-diff` page](https://github.com/tldr-pages/tldr/blob/a99a8d98d86e857ab1174804cfbecf8b170f9cd7/pages/common/git-diff.md).

File: pages/common/git-range-diff.md
Comment (xerus2000) [reply pages/common/git-range-diff.md:12]: This is very misleading.
Neither of these is "base" or "head", they are just two committishs, where the base is the common ancestor.

File: pages/common/git-range-diff.md
Comment (xerus2000) [reply pages/common/git-range-diff.md:10]: yep, I had a look at that.
tbh, if we want to be correct, the mentions of commit and branch in such pages should actually be replaced by `commit-ish` (see https://stackoverflow.com/a/23303550) since these are interchangeable.

Comment (bl-ue) [reply pages/common/git-range-diff.md:10]: Wow, you're right. We might need to change that.  Let's wait for @sbrl's opinion then.

File: pages/common/git-range-diff.md
Comment (bl-ue) [reply pages/common/git-range-diff.md:12]: Agreed.  ping @sbrl

File: pages/common/git-range-diff.md
Comment (sbrl) [reply pages/common/git-range-diff.md:10]: Heya! Sorry for the wait here. I would suggest that `ours` and `theirs` would be best used when disambiguating merges - e.g. "our commits" and "their commits". I'm not sure it really helps clarity here - I suggest changing the wording here would be the best course of action to be more specific - both as to _what_ is being referred to, and also how their are different from one another.

The man page is also singularly unhelpful:

```
       <rev1>...<rev2>
           Equivalent to passing <rev2>..<rev1> and <rev1>..<rev2>.
```

I'm unsure as how useful this example is. In theory it looks like to me that this command would be most useful when used to view the differences between 2 slightly different patches.

For example, Bob creates a patch set for project Foo to add a new feature. Bob sends the patch set to Alice, who makes some changes to the patch set with `git rebase -i` and sends them back to Bob. Bob then uses `git range-diff` to compare Alice's patch set with their own local patch set.

File: pages/common/git-range-diff.md
Comment (sbrl) [reply pages/common/git-range-diff.md:12]: I see. What about this?
```suggestion
`git range-diff {{rev_a}}...{{rev_b}}`
```

In this particular and specific instance the man page suggests that this is not a commit-ish but rather a specific revision:

```
SYNOPSIS
       git range-diff [--color=[<when>]] [--no-color] [<diff-options>]
               [--no-dual-color] [--creation-factor=<factor>]
               ( <range1> <range2> | <rev1>...<rev2> | <base> <rev1> <rev2> )

```

File: pages/common/git-range-diff.md
Comment (xerus2000) [reply pages/common/git-range-diff.md:10]: Yes, and checking the diff after a rebase is a core use-case of this command imo, that's why I want this example. But that still leaves the question of the wording open ;)

File: pages/common/git-range-diff.md
Comment (xerus2000) [reply pages/common/git-range-diff.md:12]: well, it definitely also accepts branch names, which are commitishs not revisions. I guess the man page only abbreviates that for simplicity.

reference: https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefrevisionarevision

File: pages/common/git-range-diff.md
Comment (sbrl) [reply pages/common/git-range-diff.md:10]: Oh, I see! I'm not sure I totally understand how that would work though, so I'm unsure if I'm the best person to come up with wordingg here @xerus2000. Given that's the central use-case of this command though, it might make sense to base the example around that to make it more relatable and useful. Perhaps we could reword this example to talk specifically about checking a diff after a rebase?

Comment (xerus2000) [reply pages/common/git-range-diff.md]: Tried adding examples :)

**Review by Waples — APPROVED** [review]

**Review by marchersimon — APPROVED** [review]

File: pages/common/git-range-diff.md
Comment (bl-ue) [inline pages/common/git-range-diff.md:6]: Would this work?
```suggestion
- Diff the changes between two commits:
```

Comment (xerus2000) [reply pages/common/git-range-diff.md:6]: I'm a bit wary here. Phrased like that it sounds like diff, which is what I'm trying to distinguish it from. It is not about the _changes between two commits_ but _the difference of changes within_ each commit (range).

**Comment by Waples** [comment]
bump @bl-ue

**Comment by Waples** [comment]
@bl-ue you gonna review this again or is it ready?

File: pages/common/git-range-diff.md
Comment (CleanMachine1) [reply pages/common/git-range-diff.md:6]:
```suggestion
- Diff the changes in between 2 specific commits:
```

**Comment by CleanMachine1** [comment]
We should look at the existing unresolved review bl-ue has left, I have added my input on it as well

**Comment by bl-ue** [comment]
@CleanMachine1 `stalebot ignore` is used for labels that we're ignoring because of stuff like waiting for linter updates, etc...

File: pages/common/git-range-diff.md
Comment (marchersimon) [reply pages/common/git-range-diff.md:6]: Another suggestion:
```suggestion
- Diff the changes of two specific commits:
```

Comment (CleanMachine1) [reply pages/common/git-range-diff.md:6]: @marchersimon your suggestion falls under the comment @xeruf made

File: pages/common/git-range-diff.md
Comment (CleanMachine1) [inline pages/common/git-range-diff.md:8]:
```suggestion
`git range-diff {{commit_1}}^! {{commit_2}}^!`
```

This just makes it look a little less messy

**Comment by CleanMachine1** [comment]
The tag is needed. It has been 30 days since latest reply

File: pages/common/git-range-diff.md
Comment (marchersimon) [reply pages/common/git-range-diff.md:6]: Then how about:
```suggestion
- Diff the changes made by two specific commits:
```
?

Comment (CleanMachine1) [reply pages/common/git-range-diff.md:6]: Still think between is better but still need other's opinion

Comment (sbrl) [reply pages/common/git-range-diff.md:6]: What about this then?
```suggestion
- Diff the changes with in commit within the range from `commit1` to `commit2`:
```

Comment (CleanMachine1) [reply pages/common/git-range-diff.md:6]: Yes that would work however its too long, my suggestion is still the best IMO

Comment (bl-ue) [reply pages/common/git-range-diff.md:6]: `with in`??

Comment (CleanMachine1) [reply pages/common/git-range-diff.md:6]: > Within and without should not be written with a space between with/in or with/out, according to all major dictionaries.

Comment (bl-ue) [reply pages/common/git-range-diff.md:6]: I see.  Looks like a typo @sbrl.  I assume you mean like this?
```suggestion
- Diff the changes within the range from `commit1` to `commit2`:
```

Comment (xeruf) [reply pages/common/git-range-diff.md:6]: @bl-ue you are again misunderstanding the command. There is no range created of these two commits. Only the diffs of the commits themselves are diffed.

As for between: In all suggestions so far it could be interpreted also in its time dimension, feeding the same misunderstanding as @bl-ue

Comment (xeruf) [reply pages/common/git-range-diff.md]: @sbrl also understood it wrongly.
The suggestions by @marchersimon were pretty good so I adapted them.
Don't submit any more suggestions unless you are sure you know what the command does.

**Comment by xeruf** [comment]
Ready to merge imo

File: pages/common/git-range-diff.md
Comment (bl-ue) [reply pages/common/git-range-diff.md]: I would be great if we could find an example... 😆

**Review by bl-ue — APPROVED** [review]

File: pages/common/git-range-diff.md
Comment (CleanMachine1) [inline pages/common/git-range-diff.md:16]:
```suggestion
- Diff the changes of two commit ranges, e.g. to check whether conflicts have been resolved appropriately when rebasing commits from `base_1` to `base_2`:

`git range-diff {{base_1}}..{{rev_1}} {{base_2}}..{{rev_2}}`
```

**Comment by CleanMachine1** [comment]
I'd like to dismiss simons review and get another one since its very out of date and I am not sure what his person opinion on this currently is

File: pages/common/git-range-diff.md
Comment (bl-ue) [reply pages/common/git-range-diff.md:16]: I'm not sure that this is necessary — the extra underscores render it a little less readble IMO.

Comment (marchersimon) [reply pages/common/git-range-diff.md:16]: I also think we should leave it. Since neither `e`, nor `v` is full-height, the number following it is quite readable.

**Review by marchersimon — APPROVED** [review]
LGTM, thanks @xeruf

File: pages/common/git-range-diff.md
Comment (CleanMachine1) [reply pages/common/git-range-diff.md:16]: However to keep standard, the underscores were used earlier

Comment (xeruf) [reply pages/common/git-range-diff.md:16]: I guess we should revert that, since git itself also uses no underscores, see for example:

    find /usr/share/man -name "*git*" | xargs gunzip -c | grep commit1

produces some results while the same with underscores doesn't

Comment (sbrl) [reply pages/common/git-range-diff.md:16]: Sometimes underscores (aka snake_case) make it more readable, and it other situations they don't.

In this case, given the number of items here I'm not convinced that this improves it.

File: pages/common/git-range-diff.md
Comment (sbrl) [reply pages/common/git-range-diff.md]: Oops, my suggestion meant to be `within each commit`.

File: pages/common/git-range-diff.md
Comment (CleanMachine1) [reply pages/common/git-range-diff.md:16]: feel free to ignore this then, just keeping standard since snake was used earlier

**Review by sbrl — APPROVED** [review]
This looks ok to me. This PR has been sitting around for a while and I think it's as good as it's going to get, so let's merge it as-is.

File: pages/common/git-range-diff.md
Comment (CleanMachine1) [reply pages/common/git-range-diff.md:16]: https://github.com/tldr-pages/tldr/pull/5701/files#diff-83dbb0d1c631c1f5de07d28506db0756b728f046436d4ae0524d7f05d46c2364R8

For reference to what my previous message included.

**Comment by xeruf** [comment]
Thanks :) I only check-in every few weeks on such projects since I have other more important projects my plate, with PRs like this being little side-strolls ;)

**Comment by bl-ue** [comment]
>with PRs like this being little side-strolls ;)

And they shouldn't be anything else 😉

**Pull request merged**
